Tests/SwiftMailer/MailjetTransportv31Test.php

```diff
@@ -0,0 +1,336 @@
+<?php
+
+namespace Mailjet\MailjetSwiftMailer\Tests\SwiftMailer;
+
+use PHPUnit\Framework\TestCase;
+use Mailjet\MailjetSwiftMailer\SwiftMailer\MailjetTransport;
+use Symfony\Component\Config\Definition\Processor;
+
+class MailjetTransportv31Test extends TestCase {
+
+    const MAILJET_TEST_API_KEY = 'ABCDEFG1234567';
+    const MAILJET_TEST_API_SECRET = 'ABCDEFG1234567';
+
+    /**
+     * @var \PHPUnit_Framework_MockObject_MockObject|\Swift_Events_EventDispatcher
+     */
+    protected $dispatcher;
+
+    protected function setUp() {
+        $this->dispatcher = $this->createMock('\Swift_Events_EventDispatcher');
+    }
+
+    protected function tearDown() {
+        $this->dispatcher = null;
+    }
+
+    /**
+     * Returns an instance of the transport through which test messages can be sent
+     *
+     * @return MailjetTransport
+     */
+    protected function createTransport() {
+        $clientOptions = ['url' => "api.mailjet.com", 'version' => 'v3.1', 'call' => false];
+        $transport = new MailjetTransport($this->dispatcher, self::MAILJET_TEST_API_KEY, self::MAILJET_TEST_API_SECRET, false, $clientOptions);
+        $transport->setApiKey(self::MAILJET_TEST_API_KEY);
+        $transport->setApiSecret(self::MAILJET_TEST_API_SECRET);
+        $transport->setCall(false); // Do not perform the call
+        return $transport;
+    }
+
+    public function testCanBeInstanciable() {
+        echo "Running test for SendAPI v3.1";
+        $this->assertInstanceOf(
+                MailjetTransport::class, $this->createTransport()
+        );
+    }
+
+    public function testSendTextEmail() {
+        $transport = $this->createTransport();
+        $message = new \Swift_Message('Test Subject', 'Foo bar');
+        $message
+                ->addTo('to@example.com', 'To Name')
+                ->addFrom('from@example.com', 'From Name')
+        ;
+        $message->setBody("Hello world!", 'text/plain');
+        $mailjetMessage = $transport->messageFormat->getMailjetMessage($message)['Messages'][0];
+        $result = $transport->send($message);
+        $this->assertEquals('Hello world!', $mailjetMessage['TextPart']);
+        $this->assertMessageSendable($message);
+    }
+
+    public function testSendHTMLEmail() {
+        $transport = $this->createTransport();
+        $message = new \Swift_Message('Test Subject', 'Foo bar');
+        $message
+                ->addTo('to@example.com', 'To Name')
+                ->addFrom('from@example.com', 'From Name')
+        ;
+        $message->setBody("<!DOCTYPE html>
+                <html>
+                <body>
+
+                <h1>My First Heading</h1>
+
+                <p>My first paragraph.</p>
+
+                </body>
+                </html>
+                ", "text/html");
+        $mailjetMessage = $transport->messageFormat->getMailjetMessage($message)['Messages'][0];
+        $result = $transport->send($message);
+
+        $this->assertEquals("<!DOCTYPE html>
```

Comment:
Better store it in a constant so if you update the `setBody`, you do not have to update the `assertEquals`?